examples/allocation-endpoint/README.md

```diff
@@ -0,0 +1,99 @@
+
+---
+**NOTE**
+
+The solution has not been tested with non-GKE clusters.
+
+---
+
+When creating GKE Clusters, [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) needs to be enabled. Please follow [GKE cluster setup](https://agones.dev/site/docs/installation/creating-cluster/gke/) and include workload-pool, e.g:
+
+<pre>
+gcloud container clusters create [NAME] \
+  ...
+  <b>--workload-pool=[GKE-PROJECT-ID].svc.id.goog \ </b>
+</pre>
+
+Install Agones on your cluster. You need to disable mTLS because the `agones-allocator` container will be the backend for ESP container.
+
+```
+helm upgrade my-release --install --namespace agones-system --create-namespace agones/agones \
+  --set agones.allocator.disableMTLS=true \
+  --set agones.allocator.disableTLS=true \
+  --set agones.allocator.service.http.enabled=false
+```
+
+After installing Agones, deploy [ESP](https://cloud.google.com/endpoints/docs/grpc/specify-esp-v2-startup-options) which is an envoy based proxy in front of agones `agones-alloator` container. Run the following to patch the service deployement, change the service port to ESP and add annotation to `agones-allocator` service account to impersonate GCP service account. 
+
+Replace [GKE-PROJECT-ID] in `patch-agones-allocator.yaml` with your project ID before runing the scripts.
+
+```
+kubectl patch deployment agones-allocator -n agones-system --patch-file patch-agones-allocator.yaml
+kubectl patch svc agones-allocator -n agones-system --type merge -p '{"spec":{"ports": [{"port": 443,"name":"https","targetPort":9443}]}}'
+kubectl annotate sa -n agones-system agones-allocator iam.gke.io/gcp-service-account=ae-esp-sa@[PROJECT-ID].iam.gserviceaccount.com
+```
+
+## Terraform 
+The terraform modules create resources in GCP:
+
+```
+terraform apply -var "project_id=[PROJECT-ID]" -var "ae_proxy_image=[GCR-IMAGE]:[VERSION]" -var "authorized_members=[\"serviceAccount:[SERVICE-ACCOUNT-EMAIL]\"]" -var "clusters_info=[CLUSTERS-INFO]" -var "workload-pool=[GKE-PROJECT-ID].svc.id.goog"
+```
+
+`[CLUSTERS-INFO]` is in the form of `[{\"name\":\"cluster1\",\"endpoint\":\"34.83.14.82\",\"namespace\":\"default\",\"allocation_weight\":100},{...}]` deserializing to []ClusterInfo, defined in the `server/clusterselector.go`.
+
+- The `name` is a unique randomly selected name for the cluster.
+- The `endpoint` is the `agones-allocator` external IP.
+- The `namespace` is the game server namespace.
+- The `allocation_weight` is a value between 0 and 100, which sets the relative allocation rate a cluster receives compared to other clusters. By setting weight to zero, a cluster stops receiving allocation requests.
+
+`[GCR-IMAGE]` has a default set to `gcr.io/agones-images/allocation-endpoint-proxy:latest`.
+
+`[SERVICE-ACCOUNT-EMAIL]` is the service account to be granted access the Allocation Endpoint.
```

Comment:
This is the first mention of this service account. I'm guessing that I'm supposed to create one for this purpose, but it would be nice to include a sample command to create the service account prior to this step.

Reply:
I added a note here with a link on how to create a service account. Thanks for the suggestion.